tests/Schema/BlueprintTest.php

```diff
@@ -279,6 +282,51 @@ public function test_invisible_columns(): void
         $this->assertArrayNotHasKey('name', $row);
     }
 
+    public function test_tokenList_and_search_index(): void
+    {
+        $conn = $this->getDefaultConnection();
+        $grammar = new Grammar();
+        $parentTableName = $this->generateTableName('Parent');
+        $childTableName = $this->generateTableName('Child');
+
+        $blueprint = new Blueprint($parentTableName, function (Blueprint $table) {
+            $table->create();
+            $table->uuid('pid')->primary();
+        });
+        $blueprint->build($conn, $grammar);
+
+        $blueprint = new Blueprint($childTableName, function (Blueprint $table) use ($parentTableName) {
+            $table->create();
+            $table->uuid('id');
+            $table->uuid('pid');
+            $table->string('name');
+            $table->tokenList('nameTokens', TokenizerFunction::Substring, 'name', ['language_tag' => 'en']);
+            $table->integer('number');
+            $table->interleaveInParent($parentTableName);
+            $table->primary(['pid', 'id']);
+
+            $table->fullText(['pid', 'nameTokens'])
+                ->storing(['name', 'number'])
+                ->interleaveIn($parentTableName)
+                ->partitionBy('name')
+                ->orderBy(['number' => 'desc'])
+                ->sortOrderSharding()
+                ->disableAutomaticUidColumn();
+        });
+
+        $statements = $blueprint->toSql($conn, $grammar);
+
+        $indexPrefix = Str::snake($childTableName);
+        $this->assertSame([
+            "create table `{$childTableName}` (`id` string(36) not null, `pid` string(36) not null, `name` string(255) not null, `nameTokens` tokenlist as (tokenize_substring(`name`, language_tag => 'en')) hidden, `number` int64 not null) primary key (`pid`, `id`), interleave in parent `{$parentTableName}`",
+            "create search index `{$indexPrefix}_pid_nametokens_fulltext` on `{$childTableName}`(`pid`, `nameTokens`) storing (`name`, `number`) partition by `name` order by `number` desc, interleave in `{$parentTableName}` options (sort_order_sharding=true, disable_automatic_uid_column=true)",
```

Comment:
Does the interleaved search index need to specify the parent table key as prefix of the partition?
This statement specifies a non-key column as partition.

Reply:
Parent table key is not necessary. Check out the official example which creates a similar schema.

https://cloud.google.com/spanner/docs/full-text-search/search-indexes#define_an_interleaved_search_index

Reply:
In this official example, SingerId, the key of the parent Singers table, is specified as partition.
The documentation then states

>Like interleaved tables and secondary indexes, make the key of the parent table a prefix of the PARTITION BY columns in the interleaved search index.

https://cloud.google.com/spanner/docs/full-text-search/search-indexes#interleaved_search_indexes

I cannot confirm this because I do not have an environment where I can actually test it, shouldn't the pid be specified as a partition in this test case?

Reply:
Sorry, I misread the initial comment.
I confirmed that running the test on the real spanner instance resulted in error.

```
RuntimeException: {"code":9,"message":"Search Index child_20241218_051323_671_pid_nametokens_fulltext's PARTITION BY key name does not reference index parent key column pid.","details":[]}
```

The spanner emulator is wrong it seems.
[Other bugs](https://github.com/GoogleCloudPlatform/cloud-spanner-emulator/issues/197) prevent this test from passing so execution of the schema will be limited to non-emulator for now.

This has been fixed.

Reply:
thanks!